job-server/src/main/scala/spark/jobserver/io/JobSqlDAO.scala

```diff
@@ -137,8 +138,11 @@ class JobSqlDAO(config: Config) extends JobDAO with FileCacher {
                           binaryType: BinaryType,
                           uploadTime: DateTime,
                           binBytes: Array[Byte]) {
-    // The order is important. Save the jar file first and then log it into database.
-    cacheBinary(appName, binaryType, uploadTime, binBytes)
+    val cacheOnUploadEnabled = Try(config.getBoolean("spark.jobserver.cache-on-upload")).getOrElse(true)
```

Comment:
If you have default value in application.conf then you don't need to use Try

Reply:
Done
Created default value in application.conf and removed Try from JobSqlDAO